typespec-extension/src/code-model-builder.ts

```diff
@@ -1019,76 +999,67 @@ export class CodeModelBuilder {
 
   private _armApiVersionParameter?: Parameter;
 
-  private processParameter(op: CodeModelOperation, param: HttpOperationParameter, clientContext: ClientContext) {
+  private processParameter(
+    op: CodeModelOperation,
+    param: SdkQueryParameter | SdkPathParameter | SdkHeaderParameter,
+    clientContext: ClientContext,
+  ) {
     if (clientContext.apiVersions && isApiVersion(this.sdkContext, param)) {
       // pre-condition for "isApiVersion": the client supports ApiVersions
       if (this.isArm()) {
         // Currently we assume ARM tsp only have one client and one api-version.
         // TODO: How will service define mixed api-versions(like those in Compute RP)?
-        const apiVersion = this.apiVersion?.value;
+        const apiVersion = this.apiVersion;
         if (!this._armApiVersionParameter) {
           this._armApiVersionParameter = this.createApiVersionParameter(
             "api-version",
-            param.type === "query" ? ParameterLocation.Query : ParameterLocation.Path,
+            param.kind === "query" ? ParameterLocation.Query : ParameterLocation.Path,
             apiVersion,
           );
           clientContext.addGlobalParameter(this._armApiVersionParameter);
         }
         op.addParameter(this._armApiVersionParameter);
       } else {
-        const parameter = param.type === "query" ? this.apiVersionParameter : this.apiVersionParameterInPath;
+        const parameter = param.kind === "query" ? this.apiVersionParameter : this.apiVersionParameterInPath;
         op.addParameter(parameter);
         clientContext.addGlobalParameter(parameter);
       }
-    } else if (this.isSubscriptionId(param)) {
+    } else if (param.kind === "path" && param.onClient && this.isSubscriptionId(param)) {
       const parameter = this.subscriptionIdParameter(param);
       op.addParameter(parameter);
       clientContext.addGlobalParameter(parameter);
```

Comment:
You need to add logic to handle `param.onClient`. (part of it need to be merged with `subscriptionId`)

If TCGC gives `param.onClient`, it could be `clientContext.addGlobalParameter(parameter)`.
```
if (param.onClient) {
  clientContext.addGlobalParameter(parameter);
}
```

This applies to all parameters. So this logic probably near the end of this function.

This is not easy to test right now. It may depend on https://github.com/Azure/typespec-azure/issues/914

Reply:
Does `onClient` only applies to header/query/path params? Or it also applies to body parameters? From the issue seems it also applies to body parameters, but the design is not finalized yet.

For appying to header/query/path params case, I am a bit unusre about the requirement and use case (besides subscription id and api-version). I tried to add logics in this commit. Please correct me if I understand wrongly.

Reply:
I don't think it could be in body.

Use case be that e.g. a header (or path segment) that going to be same for the whole client.

E.g. `@header region` that going to be same e.g. "WESTUS" / "EASTUS" for all API calls in the client.

Reply:
Could you double check whether `op.addParameter(parameter);` need to be called in both case (for double confirm, do we call `addParameter` on current api-version and subscriptionId?

Other part of the commit LGTM.

Reply:
Thanks for catching. We call `op.addParameter()` currently for subscription id and api-version. https://github.com/Azure/autorest.java/blob/main/typespec-extension/src/code-model-builder.ts#L1048

We may need to call this in both cases. I will update code and test.

Reply:
https://github.com/Azure/autorest.java/issues/2920 - created an issue for Java.
depends on TCGC issue: https://github.com/Azure/typespec-azure/issues/1391